src/getValidators.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  Address,
+  Hex,
+  PublicClient,
+  decodeFunctionData,
+  getAbiItem,
+  getFunctionSelector,
+} from 'viem';
+import { rollupCreator, upgradeExecutor } from './contracts';
+import { rollupAdminLogicABI, safeL2ABI } from './abi';
+
+const createRollupABI = getAbiItem({ abi: rollupCreator.abi, name: 'createRollup' });
+const createRollupFunctionSelector = getFunctionSelector(createRollupABI);
+
+const setValidatorABI = getAbiItem({ abi: rollupAdminLogicABI, name: 'setValidator' });
+const setValidatorFunctionSelector = getFunctionSelector(setValidatorABI);
+
+const executeCallABI = getAbiItem({ abi: upgradeExecutor.abi, name: 'executeCall' });
+const upgradeExecutorExecuteCallFunctionSelector = getFunctionSelector(executeCallABI);
+
+const execTransactionABI = getAbiItem({ abi: safeL2ABI, name: 'execTransaction' });
+const safeL2FunctionSelector = getFunctionSelector(execTransactionABI);
+
+const ownerFunctionCalledEventAbi = getAbiItem({
+  abi: rollupAdminLogicABI,
+  name: 'OwnerFunctionCalled',
+});
+
+function getValidatorsFromFunctionData<
+  TAbi extends (typeof createRollupABI)[] | (typeof setValidatorABI)[],
+>({ abi, data }: { abi: TAbi; data: Hex }) {
+  const { args } = decodeFunctionData({
+    abi,
+    data,
+  });
+  if (typeof args === 'undefined') {
+    return [];
+  }
+  return args;
+}
+
+function setValidators(acc: Set<Address>, input: Hex) {
+  const [validators, states] = getValidatorsFromFunctionData({
+    abi: [setValidatorABI],
+    data: input,
+  });
+
+  validators.forEach((validator, i) => {
+    const isAdd = states[i];
+    if (isAdd) {
+      acc.add(validator);
+    } else {
+      acc.delete(validator);
+    }
+  });
+
+  return acc;
+}
+
+type GetValidatorsParams = {
+  /** Address of the rollup we're getting list of validators from */
+  rollupAddress: Address;
+};
+type GetValidatorsReturnType = {
+  /** If logs contain unknown signature, validators list is not guaranteed to be complete */
+  isComplete: boolean;
+  /** List of validators for the given rollup */
+  validators: Address[];
+};
+
+/**
+ *
+ * @param {PublicClient} client - The chain Viem Public Client
+ * @param {GetValidatorsParams} GetValidatorsParams {@link GetValidatorsParams}
+ * @param GetValidatorsParams.rollupAddress
+ *
+ * @returns Promise<{@link GetValidatorsReturnType}> - The validators with a flag, whether or not we guarantee that all validators were fetched
+ *
+ * @example
+ * const { isComplete, validators } = getValidators(client, { rollupAddress: '0xc47dacfbaa80bd9d8112f4e8069482c2a3221336' });
+ *
+ * if (isComplete) {
+ *   // Validators were all fetched properly
+ * } else {
+ *   // Some validators might be missing
+ * }
+ */
+export async function getValidators(
+  client: PublicClient,
+  { rollupAddress }: GetValidatorsParams,
```

Comment:
I think this is more in line with our other APIs?

```suggestion
  { rollup }: GetValidatorsParams,
```